In [41]:
# bibliotecas necessárias
import pandas as pd

# criando objeto com caminho e carregando arquivo csv
path = '/home/robson/repositorios/bike-manufacturing-company/data-lake/landing/Person.Person.csv'
loaded_csv = pd.read_csv(path, sep=';')

# criando dataframe pandas
df = pd.DataFrame(loaded_csv)

#### Análise estrutural

In [42]:
# verificando estrutura do dataframe (rows, columns)
df.shape

(19972, 13)

In [43]:
# verificando tipos de dados
df.dtypes

BusinessEntityID          int64
PersonType               object
NameStyle                 int64
Title                    object
FirstName                object
MiddleName               object
LastName                 object
Suffix                   object
EmailPromotion            int64
AdditionalContactInfo    object
Demographics             object
rowguid                  object
ModifiedDate             object
dtype: object

In [44]:
# analisando valores unicos de cada coluna nao-numerica
df.describe(exclude='number')

PersonType Title FirstName MiddleName LastName Suffix  \
count       19972  1009     19972      11473    19972     53   
unique          6     6      1018         71     1206      6   
top            IN   Mr.   Richard          A     Diaz    Jr.   
freq        18484   577       103       1319      211     33   

                                    AdditionalContactInfo  \
count                                                  10   
unique                                                 10   
top     <AdditionalContactInfo xmlns="http://schemas.m...   
freq                                                    1   

                                             Demographics  \
count                                               19972   
unique                                              19114   
top     <IndividualSurvey xmlns="http://schemas.micros...   
freq                                                  853   

                                     rowguid             ModifiedDate  
count                                  19972                    19972  
unique                                 19972                     1285  
top     92C4279F-1207-48A3-8448-4636514EB7E2  2013-07-31 00:00:00.000  
freq                                       1                      136

In [45]:
df.describe(include='number')

BusinessEntityID  NameStyle  EmailPromotion
count      19972.000000    19972.0    19972.000000
mean       10763.079411        0.0        0.630082
std         5814.133272        0.0        0.781433
min            1.000000        0.0        0.000000
25%         5798.750000        0.0        0.000000
50%        10791.500000        0.0        0.000000
75%        15784.250000        0.0        1.000000
max        20777.000000        0.0        2.000000

#### Análise qualitativa

In [46]:
# verificando existencia de dados nulos por coluna
df.isnull().sum()

BusinessEntityID             0
PersonType                   0
NameStyle                    0
Title                    18963
FirstName                    0
MiddleName                8499
LastName                     0
Suffix                   19919
EmailPromotion               0
AdditionalContactInfo    19962
Demographics                 0
rowguid                      0
ModifiedDate                 0
dtype: int64

### Conclusão:

1- Coluna "Title" contendo 18963 campos vazios. Como é um campo que não terá influência nas análises, será descartada a coluna.

2- Nomes pessoais na tabela separados por "FirstName", "MiddleName" e "LastName", tendo ainda o campo "Suffix". Apesar de não ser o ideal (em algumas situações o campo "MiddleName" ficará vazio), resolvi deixar as 3 colunas como estão, não excluindo a coluna "MiddleName" e nem mesclando as colunas em uma única coluna "Name". Sera inserido "uninformed" nos campos vazios

3- Coluna "AdditionalContactInfo com 19962 campos vazios. Será descartada a coluna, pois não serão feitas análises com tal informação. 

4- Colunas "BusinessEntityID", "NameStyle" e "EmailPromotion" do tipo "int", sendo prudente transformá-las em "string".